gpflow/utilities/utilities.py

```diff
@@ -48,6 +49,13 @@ def set_trainable(model: tf.Module, flag: bool):
     """
     Set trainable flag for all `tf.Variable`s and `gpflow.Parameter`s in a module.
     """
+    if not tf.executing_eagerly():
```

Comment:
Honestly! This is madness :) you can put this message in every single function of GPflow, just because things might change with `tf.function`. I do not know any case where this can happen - none!

Reply:
@joelberkeley-pio ^^^

Reply:
will remove